### PR TITLE
Fix Windows build

### LINKED ISSUE #57: Update CMake build directory layout to ease running binary in build directory on Windows
Windows don't have RPATH, then, if the dll is not in the same directory than the executable the only option to launch it is to modify the PATH environment variable.

We can modify PATH environment when running tests in CMake but:
- This add some Windows specific code
- The only way to launch example/tests/benchmark is to use ctest

Another option is to use the install layout in the build directory:

├── bin
│   ├── dll
│   └── exe
└── lib
    ├── a
    ├── dylib
    ├── lib
    └── so

Since dll and exe are in the same directory, there is no execution problem.
Thanks to the RPATH, this layout is also working on GNU/Linux and MacOS.

Also, `windows_dll_manager.py` can find the bin directory to find the dll without using PROXSUITE_NLP_WINDOWS_DLL_PATH environment variable.

### LINKED ISSUE #58: Windows support
We need to fix building the library on Windows along with a CI ~~for some ungodly reason~~

### PR DESCRIPTION
**Pull request by jorisv**
- [x] build error
- [x] CMake error
- [x] template instantiation export
- [x] update build layout
- [x] use standard jrl cmake module
- [x] add CI
- [x] publish CI
- [x] fix extern template ClangCL
- [x] fix windows manager dll paths

Close #57 

Resolves #58

### CODE REVIEW

**Review by ManifoldFR — APPROVED** [review]
I look upon this PR with both awe and sheer horror at what needed to be done to get it working on the cursed platform known as Windows.

Looks good to me

**Review by ManifoldFR — APPROVED** [review]

File: include/proxsuite-nlp/prox-solver.hxx [hunk 1]
```diff
@@ -456,7 +456,7 @@ void ProxNLPSolverTpl<Scalar>::innerLoop(Workspace &workspace,
                      dphi0,
                      results.merit,
                      phi_new - phi0,
-                     static_cast<size_t>(results.al_iters + 1)};
+                     static_cast<unsigned long>(results.al_iters + 1)};
```
Comment (ManifoldFR) [inline include/proxsuite-nlp/prox-solver.hxx:459]: Wait, I think we should maybe change the logger to use `size_t` instead.

Comment (jorisv) [reply include/proxsuite-nlp/prox-solver.hxx:459]: Ok, I will change it

**Review by ManifoldFR — APPROVED** [review]
Thanks a lot @jorisv, and further thanks for the explanations on what the heck is going on

@stephane-caron we will be doing a Windows release soon ;)

**Pull request merged**